doc/run.md

```diff
@@ -364,6 +364,81 @@ packet egresses from the sender.
+     ```bash
+     ./build/app/TxSt40PipelineSample \
+             --p_port 0000:af:01.0 \
+             --p_tx_ip 239.1.1.1 \
+             --p_sip 192.168.96.2 \
+             --udp_port 20000 \
+             --payload_type 40 \
+             --sessions_cnt 1 \
+             --tx_url /tmp/anc_payload.bin \
+             --ext_frame
+     ```
+
+Key behaviors:
+
+- `--ext_frame` keeps the sample in "user supplied frame" mode, matching how most applications integrate the pipeline API.
+- When `--multi_inc_addr` is present, each session increments the destination multicast address instead of the UDP port so you can fan out multiple streams on a single port.
+- Destination MAC overrides (`--p_tx_dst_mac`/`--r_tx_dst_mac`) work the same way as the other apps if you need explicit L2 addressing.
+
+#### 5.4.2. Rx sample workflow
+
+Run the RX sample with matching network parameters. Enabling `--rx_dump` writes one human-readable metadata report per frame to `--rx_url` (`foo.txt_0`, `_1`, … when multiple sessions are active):
+
+```bash
+./build/app/RxSt40PipelineSample \
+        --p_port 0000:af:01.0 \
+        --p_rx_ip 239.1.1.1 \
+        --p_sip 192.168.96.2 \
+        --udp_port 20000 \
+        --payload_type 40 \
+        --sessions_cnt 1 \
+        --rx_url /tmp/anc_dump.txt \
+        --rx_dump
+```
+
+The log stream shows per-frame metadata counts plus the number of user data words (UDW) that landed in the buffer. The dump file records DID/SDID/line info followed by the raw payload bytes so you can diff TX/RX easily.
+
+#### 5.4.3. Troubleshooting and USDT hooks
+
+- Use `--rx_dump`/`--tx_url` first to confirm that metadata makes the round trip.
+- When `st40_total_pkts` exceeds 1, it means the payload exceeded 255 bytes; trim the input file or lower `--sessions_cnt` on slow links.
+- Attach `bpftrace` to the USDT probes to see real-time pipeline events:
+    ```bash
+    sudo BPFTRACE_STRLEN=200 bpftrace -e '
+    usdt::st40p:rx_frame_available {
+        printf("%s rx%u: frame %u meta=%u\n", strftime("%H:%M:%S", nsecs), arg0, arg1, arg2);
+    }
+    usdt::st40p:rx_frame_put {
+        printf("%s rx%u: frame %u returned\n", strftime("%H:%M:%S", nsecs), arg0, arg1);
+    }
+    usdt::st40p:rx_frame_dump {
+        printf("%s rx%u: dump meta=%u bytes=%u -> %s\n",
+                     strftime("%H:%M:%S", nsecs), arg0, arg2, arg3, str(arg1));
+    }
+    ' -p $(pidof RxSt40PipelineSample)
+    ```
+- `usdt::sys:log_msg` mirrors the built-in status dump; attach to it when you want the sample to spit out scheduler stats without touching its log level.
+
+These executables are the quickest way to validate port configuration, payload sizing, and USDT instrumentation before embedding the ST40 pipeline API into your production app.
+
```

Comment:
i also don't think that should be added, the usdt is described in detail in usdt.md, i don't think adding it to the 
**run.md** describing how to run the application is a place for it

Reply:
Amazing **ദ്ദി(˵ •̀ ᴗ - ˵ ) ✧**